tests/test_engine.py

```diff
@@ -420,12 +420,16 @@ def _assert_signals_caught(self, run: CrawlerRun):
 
     @defer.inlineCallbacks
     def test_close_downloader(self):
-        e = ExecutionEngine(get_crawler(TestSpider), lambda _: None)
+        crawler = get_crawler(TestSpider)
+        yield crawler.crawl()
+        e = ExecutionEngine(crawler, lambda _: None)
         yield e.close()
```

Comment:
1. Why is this test called test_close_downloader?
2. Why is the engine closed, but the crawler isn't? Would it make more sense to pick engine from the crawler, and the close the crawler?
3. tbh, I don't quite get what is this test testing

Reply:
> Why is this test called test_close_downloader?

Initially there were three tests added in #1423: `test_close_downloader`, `test_close_spiders_downloader` and `test_close_engine_spiders_downloader` which called and tested different open/close things, then the last two were removed when `ExecutionEngine.open_spiders` was removed.

> tbh, I don't quite get what is this test testing

I guess it just tests that `ExecutionEngine.close()` doesn't produce exceptions.